include/deal.II/base/patterns.h

```diff
@@ -149,6 +149,14 @@ namespace Patterns
   std::unique_ptr<PatternBase> pattern_factory (const std::string &description);
 
   /**
+   * Escape the string @p input for the specified @p style so that characters
+   * will appear as intended. For example, characters like _ can not be
+   * written as is in LateX and have to be escaped as \_.
+   */
+  std::string escape(const std::string &input, const PatternBase::OutputStyle style);
```

Comment:
Why put this into the `Patterns` namespace?

Reply:
I think this would be better placed into the `ParameterHandler` class itself -- which incidentally also has a `OutputStyle` enum. Or, if you want it to be a free function, place it alongside the `mangle`/`demangle` functions.

Reply:
> Why put this into the Patterns namespace?
It needs to be visible for Patterns and ParameterHandler but Patterns currently doesn't know anything anything about ParameterHandler and I did not want to introduce a dependency that way.

> Or, if you want it to be a free function, place it alongside the mangle/demangle functions.

mangle/demangle are not free functions but static members of ParameterHandler. I am happy to move it there if you prefer.

Reply:
Yes, static member is good. I don't see a need for anyone to call these functions from anywhere else but inside `ParameterHandler`.

Reply:
It is used inside ``Pattern`` so I can not make it a private static function like ``mangle`` without major reshuffling and including ``parameter_handler.h`` inside pattern. Is this really what you want?

Reply:
I just wanted it out of the public header file since it's an implementation detail. If you can't move it to the `.cc` as a function in an anonymous namespace, then move it into the `internal` namespace.